source/Handlebars/LayoutViewModel.cs

```diff
@@ -0,0 +1,94 @@
+using System;
+using System.Collections;
+using System.Linq;
+using HandlebarsDotNet.Compiler;
+using HandlebarsDotNet.Iterators;
+using HandlebarsDotNet.MemberAccessors;
+using HandlebarsDotNet.ObjectDescriptors;
+using HandlebarsDotNet.PathStructure;
+
+namespace HandlebarsDotNet
+{
+    internal class LayoutViewModel
+    {
+        private static readonly ChainSegment _bodyChainSegment = ChainSegment.Create("body");
+        private readonly string _body;
+        private readonly object _value;
+        private readonly ObjectDescriptor _valueDescriptor;
+
+        public LayoutViewModel(string body, object value)
+        {
+            _body = body;
+            _value = value;
+            _valueDescriptor = ObjectDescriptor.Create(value);
+        }
+
+        internal class DescriptorProvider: IObjectDescriptorProvider
+        {
+            private static readonly object[] _bodyProperties = { _bodyChainSegment };
+            private static readonly Type _type = typeof(LayoutViewModel);
```

Comment:
Rename to `Type`: `static readonly` fields should be named starting with capital letter.